microcosm_pubsub/tests/test_producer.py

```diff
@@ -196,3 +202,62 @@ def bar(self):
     func()
 
     assert_that(graph.sns_producer.sns_client.publish.call_count, is_(equal_to(1)))
+
+
+def test_deferred_batch_production():
+    """
+    Deferred production waits until the end of a block and publishes all
+    messages in one MessageBatchSchema
+
+    """
+    def loader(metadata):
+        return dict(
+            sns_topic_arns=dict(
```

Comment:
It would be really nice to have the validation that requires the sns topic arn not blow up in test mode.